### DISCUSSION TITLE
Documentation for Gulp Setup

### DISCUSSION THREAD

**Discussion by joshmedeski** [tailwindlabs/tailwindcss#5471] [Ideas]
Hey team, I see you're building v3 of the documentation, how exciting!

I think adding documentation for setting up a project with [Gulp](https://gulpjs.com) would be valuable.

Here's an example of a project I'm using with WordPress and Timber

```js
const { dest, src, series, watch: gulpWatch } = require("gulp");
const browserSync = require("browser-sync");
const postcss = require("gulp-postcss");
const server = browserSync.create();

function css() {
  return src("./global.css").pipe(postcss()).pipe(dest("./static"));
}

function reload(done) {
  server.reload();
  done();
}

function serve(done) {
  server.init({
    proxy: "localhost:8000",
  });
  done();
}

const watch = () => {
  gulpWatch("./**/*.twig", series(css, reload));
  gulpWatch("./**/*.php", series(css, reload));
  gulpWatch("./global.css", series(css, reload));
  gulpWatch("./tailwind.config.js", series(css, reload));
};

exports.default = series(css, serve, watch);
exports.build = css;
```

If you'd like me to break this down into full-on instructions for the documentation page I'm happy to contribute.

BTW thanks for making TailwindCSS, it's a game-changer for me 😄

**Comment by benlilley** [comment]
I think this would also be helpful as I'm running into a few challenges. In particular getting the new JIT method in 3.0 to work with `gulp watch` is being problematic for me. HTML changes do not cause the correct generation of CSS files.

**Reply by joshmedeski** [reply]
Make sure your [`tailwind.config.js` config file](https://tailwindcss.com/docs/configuration) is using the new `content` key and is properly matching the path to your HTML files.

**Reply by benlilley** [reply]
Yep, I don't want to hijack this ticket, but I can confirm that all works fine if I use the Tailwind CLI to watch the files, I get the expected behaviour but not with gulp. I wonder if it's [related to this](https://github.com/tailwindlabs/tailwindcss/discussions/4710#discussioncomment-895804), either way I have some threads to pull and hopefully will get there.

But still agree some docs for gulp to get the best out of Tailwind would be great!

**Reply by joshmedeski** [reply]
Also check the `postcss` version used by `gulp-postcss` is the same version that TailwindCSS expects. Maybe that package is using an older version of `postcss` that doesn't fully work with the new TailwindCSS.
